src/163400.c

```diff
@@ -81,7 +81,37 @@ void filemenu_set_cursor_alpha(s32 arg0) {
     filemenu_cursorGoalAlpha2 = arg0;
 }
 
-INCLUDE_ASM(s32, "163400", filemenu_set_cursor_goal_pos);
+void filemenu_set_cursor_goal_pos(s32 windowID, s32 posX, s32 posY) {
+    Window* window = &gWindows[windowID];
+
+    if (D_80249BB0 != 0
+            || get_game_mode() == 0xF
+            || get_game_mode() == 0xD) {
+        if (D_80249BB0 != 0) {
+            s32 i;
+
+            for (i = 0x2C; i < 0x40; i++) {
+                Window* window = &gWindows[i];
+                s8 parent = window->parent;
+
+                if ((parent == -1 || parent == 0x2C) && (window->flags & 8)) {
+                    break;
+                }
+            }
+            if (i >= 0x40) {
```

Comment:
```suggestion
            if (i >= ARRAY_COUNT(gWindows)) {
```

Reply:
I can do array count here, but it seems like gWindows has subdivisions within it for each range that probably need an enum.

Reply:
yeah or some defines. but I think it's safe to use this for the end of the array. Or it's multiple arrays..hm